compiler/rustc_middle/src/ty/context.rs

```diff
@@ -384,6 +384,10 @@ impl<'tcx> Interner for TyCtxt<'tcx> {
         self.is_conditionally_const(def_id)
     }
 
+    fn alias_has_const_conditions(self, def_id: DefId) -> bool {
+        self.is_conditionally_const(def_id)
```

Comment:
given that we use `is_conditionally_const` via separate functions, maybe `debug_assert` the `DefKind` here